editor/src/components/canvas/canvas-utils.ts

```diff
@@ -2180,3 +2189,69 @@ export function canvasPanelOffsets(): {
     right: inspector?.clientWidth ?? 0,
   }
 }
+
+export function projectContentsSameForRefreshRequire(
+  oldProjectContents: ProjectContentTreeRoot,
+  newProjectContents: ProjectContentTreeRoot,
+): boolean {
+  if (oldProjectContents === newProjectContents) {
+    // Identical references, so the imports are the same.
+    return true
+  } else {
+    for (const [filename, oldProjectTree] of Object.entries(oldProjectContents)) {
+      const newProjectTree = newProjectContents[filename]
+      // If the file can't be found in the other tree, the imports are not the same.
+      if (newProjectTree == null) {
```

Comment:
I think because we use `ProjectContentTreeRootKeepDeepEquality` we could also have a check here for `if (oldProjectTree === newProjectTree)`

Reply:
Improvement added in 112d15b20.